packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1917,25 +1920,42 @@ abstract class BuildContext {
   /// render object is usually short.
   Size get size;
 
+  /// Registers this build context with [ancestor] such that when
+  /// [ancestor]'s widget changes this build context is rebuilt.
+  ///
+  /// This method is deprecated. Please use [dependOnInheritedElement] instead.
+  @Deprecated('Use dependOnInheritedElement instead.')
+  InheritedWidget inheritFromElement(InheritedElement ancestor, { Object aspect });
+
   /// Registers this build context with [ancestor] such that when
   /// [ancestor]'s widget changes this build context is rebuilt.
   ///
   /// Returns `ancestor.widget`.
   ///
   /// This method is rarely called directly. Most applications should use
-  /// [inheritFromWidgetOfExactType], which calls this method after finding
+  /// [dependOnInheritedWidgetOfExactType], which calls this method after finding
   /// the appropriate [InheritedElement] ancestor.
   ///
-  /// All of the qualifications about when [inheritFromWidgetOfExactType] can
+  /// All of the qualifications about when [dependOnInheritedWidgetOfExactType] can
   /// be called apply to this method as well.
-  InheritedWidget inheritFromElement(InheritedElement ancestor, { Object aspect });
+  InheritedWidget dependOnInheritedElement(InheritedElement ancestor, { Object aspect });
 
   /// Obtains the nearest widget of the given type, which must be the type of a
   /// concrete [InheritedWidget] subclass, and registers this build context with
   /// that widget such that when that widget changes (or a new widget of that
   /// type is introduced, or the widget goes away), this build context is
   /// rebuilt so that it can obtain new values from that widget.
   ///
+  /// This method is deprecated. Please use [dependOnInheritedWidgetOfExactType] instead.
+  @Deprecated('Use dependOnInheritedWidgetOfExactType instead.')
+  InheritedWidget inheritFromWidgetOfExactType(Type targetType, { Object aspect });
```

Comment:
What's the status of extension methods these days? Can we move it into a `DeprecatedBuildContext` extension or some such, so the old methods do not pollute the interface?

Reply:
According to https://github.com/dart-lang/sdk/blob/master/CHANGELOG.md extension methods are in preview.

Moreover as some deprecated methods can not be implemented just by calling the new methods there could be some problems to have a method body that works with all implementation of `BuildContext`.

Reply:
Also, for the time being we do not want to use extension methods in the framework.

Reply:
@a14n 

> deprecated methods can not be implemented just by calling the new methods

Can they be implemented without calling the new methods?

@goderbauer

> we do not want to use extension methods

Why would you not want to use them in this particular case?

Reply:
>Can they be implemented without calling the new methods?

The current implementation of those deprecated methods (in `Element`) uses private fields. An implementation of those methods in an extension will have to access to those private fields (to be able to also use new methods). But this would lead to issues for instance in the context of unit test if you mock Element.

Even if we can come up with a workaround I don't thing it's worth to add complexity only to handle a temporary period of deprecation.